packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -816,6 +830,26 @@ class ScriptTransformer {
 
     return this._config.transform.length !== 0 && !isIgnored;
   }
+
+  private getScriptCacheKey = (
+    fileSource: undefined | string | null,
+    filename: string,
+    instrument: boolean,
+    transformOptions: any,
+  ) => {
+    let fileContent = fileSource ?? this._cacheFS.get(filename);
+    if (fileContent == null) {
+      fileContent = fs.readFileSync(filename, 'utf8');
```

Comment:
I'm worried about perf here. I'm not too familiar with our transform code, but I think the current layered caching system is as follows:

1. Check memory cache using key from `getScriptCacheKey`, which can be done only based on an `mtime stat`.
2. If miss, load file content from disk
3. Check disk cache using key from `getCacheKey`, which takes file content as an argument.

You're trying to include `getCacheKey` already in the first `getScriptCacheKey` check, but to do that you need to load the file a 100% of the time unless it was already loaded and provided to the transformer. I think we usually do that so maybe it's fine. @simenb do you know why `fileSource` is even optional to `transform`? It doesn't seem to ever not be provided in our code.

Reply:
If we do decide it's fine to do this, then I wonder: Do we even need the multiple cache key calculation  functions, or can it all be the same since we always have both metadata and content available for the calculation anyway

Reply:
> I think we usually do that so maybe it's fine

Yeah that's what I saw while debugging so I figured it would be fine.

> do you know why fileSource is even optional to transform?

I need to double check but while debugging I do believe I saw fileSource be undefined in some cases

Reply:
yeah, I think we should avoid reading the file in the first check

Reply:
I think reading the file is necessary to get the cache key from the transformer which requires the file contents to create the hash